bundles/org.openhab.core.sitemap/src/main/java/org/openhab/core/sitemap/Widget.java

```diff
@@ -0,0 +1,185 @@
+     * method should not return null.
+     *
+     * @return item, or null if no item defined for the widget
+     */
+    @Nullable
+    String getItem();
+
+    /**
+     * Sets the widget item.
+     *
+     * @param item
+     */
+    void setItem(@Nullable String item);
+
+    /**
+     * Get widget label.
+     *
+     * @return label
+     */
+    @Nullable
+    String getLabel();
+
+    /**
+     * Set widget label.
+     *
+     * @param label
+     */
+    void setLabel(@Nullable String label);
+
+    /**
+     * Get widget icon.
+     *
+     * @return icon
+     */
+    @Nullable
+    String getIcon();
+
+    /**
+     * Set widget icon.
+     *
+     * @param icon
+     */
+    void setIcon(@Nullable String icon);
+
+    /**
+     * Get the widget icon rules. This method should return a modifiable list, allowing updates to the icon rules.
+     *
+     * @return icon rules
+     */
+    List<Rule> getIconRules();
+
+    /**
+     * Replace the widget icon rules with a new list of icon rules.
+     *
+     * @param iconRules
+     */
+    void setIconRules(List<Rule> iconRules);
+
+    /**
+     * True if the widget icon is static, false otherwise.
+     *
+     * @return static icon
+     */
+    boolean isStaticIcon();
+
+    /**
+     * Set to true if the widget icon is static.
+     *
+     * @param staticIcon
+     */
+    void setStaticIcon(@Nullable Boolean staticIcon);
+
+    /**
+     * Get the widget label color rules. This method should return a modifiable list, allowing updates to the label
+     * color rules.
+     *
+     * @return label color rules
+     */
+    List<Rule> getLabelColor();
+
+    /**
+     * Replace the widget label color rules with a new list of label color rules.
+     *
+     * @param labelColor
+     */
+    void setLabelColor(List<Rule> labelColor);
+
+    /**
+     * Get the widget value color rules. This method should return a modifiable list, allowing updates to the value
+     * color rules.
+     *
+     * @return value color rules
+     */
+    List<Rule> getValueColor();
+
+    /**
+     * Replace the widget value color rules with a new list of value color rules.
+     *
+     * @param valueColor
+     */
+    void setValueColor(List<Rule> valueColor);
+
+    /**
+     * Get the widget icon color rules. This method should return a modifiable list, allowing updates to the icon
+     * color rules.
+     *
+     * @return icon color rules
+     */
+    List<Rule> getIconColor();
+
+    /**
+     * Replace the widget icon color rules with a new list of icon color rules.
+     *
+     * @param iconColor
+     */
+    void setIconColor(List<Rule> iconColor);
+
+    /**
+     * Get the widget visibility rules. This method should return a modifiable list, allowing updates to the visibility
+     * rules.
+     *
+     * @return visibility rules
+     */
+    List<Rule> getVisibility();
```

Comment:
Maybe rather getVisibilityRules and setVisibilityRules ?

Reply:
See above